pkg/odo/cli/delete/component.go

```diff
@@ -67,25 +67,44 @@ func (o *ComponentOptions) deleteNamedComponent() error {
 
 // deleteDevfileComponent deletes a component defined by the devfile in the current directory
 func (o *ComponentOptions) deleteDevfileComponent() error {
+	devfileObj := o.EnvSpecificInfo.GetDevfileObj()
+	componentName := o.EnvSpecificInfo.GetName()
+
 	// TODO: Print all the resources that will be deleted.
+	log.Info("The following components will be deleted: ")
+	log.Printf("Devfile Component: %v", componentName)
+	log.Printf("Kubernetes Resources:")
+	resources, err := o.clientset.DeleteClient.ListKubernetesComponents(devfileObj, filepath.Dir(o.EnvSpecificInfo.GetDevfilePath()))
+	if err != nil {
+		return fmt.Errorf("Failed to gather resources for deletion: %w", err)
+	}
+	for _, resource := range resources {
+		fmt.Println("\t", resource.GetKind(), ":", resource.GetName())
+	}
 
-	componentName := o.EnvSpecificInfo.GetName()
+	if o.forceFlag || ui.Proceed("Are you sure you want to delete these resources?") {
+		// delete innerloop resources
+		spinner := log.Spinner("Deleting Devfile component")
+		defer spinner.End(false)
```

Comment:
As there is no return statement before the `spinner.End(true)`, you don't need this `defer`